simpleclient_hotspot/src/main/java/io/prometheus/client/hotspot/StandardExports.java

```diff
@@ -101,6 +99,72 @@ public StandardExports() {
     return mfs;
   }
 
+  static Long callLongGetter(String getterName, Object obj) throws NoSuchMethodException {
+    Method method = obj.getClass().getMethod(getterName);
+    return (Long) callMethod(method, obj).getValue();
+  }
+
+  /**
+   * Attempts to call a method either directly or via one of the implemented interfaces.
+   */
+  static ReturnValue callMethod(Method method, Object obj) {
+    try {
+      method.setAccessible(true);
+      return new ReturnValue(method.invoke(obj));
+    } catch (Exception e) {
+      LOGGER.log(Level.FINE,
+          "Invocation failed on " + method.getDeclaringClass().getName() + "." + method.getName(),
+          e);
+    }
+
+    for (Class<?> clazz : method.getDeclaringClass().getInterfaces()) {
```

Comment:
This is the "recursively visit all implemented interfaces" part.

A Method object refers to a specific class. So the first invocation of callMethod() will happen with method === SomeConcreteClass.getProcessCpuTime(), which will fail under Java 9. We then need to iterate over all interfaces implemented by SomeConcreteClass and eventually invoke callMethod() with method === UnixOperatingSystemMXBean.getProcessCpuTime(), which will then succeed.